File: 2026/05-May/17-Sunday/solution.py
class Solution:
    def canReach(self, arr, start):

        if start < 0 or start >= len(arr) or arr[start] < 0:
            return False

        if arr[start] == 0:
            return True

        jump = arr[start]
        arr[start] *= -1

        return (self.canReach(arr, start + jump) or
                self.canReach(arr, start - jump))

File: 2026/05-May/17-Sunday/solution.cs
public class Solution {
    public bool CanReach(int[] arr, int start) {

        if(start < 0 || start >= arr.Length || arr[start] < 0)
            return false;

        if(arr[start] == 0)
            return true;

        int jump = arr[start];
        arr[start] *= -1;

        return CanReach(arr, start + jump) ||
               CanReach(arr, start - jump);
    }
}
